bee-network/src/conns/manager.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    interaction::events::InternalEventSender,
+    peers::{BannedAddrList, BannedPeerList, PeerInfo, PeerList, PeerManager, PeerRelation, PeerState},
+    transport::build_transport,
+    Multiaddr, PeerId, ShortId,
+};
+
+use super::{errors::Error, Origin};
+
+use bee_runtime::{node::Node, shutdown_stream::ShutdownStream, worker::Worker};
+
+use async_trait::async_trait;
+use futures::prelude::*;
+use lazy_static::lazy_static;
+use libp2p::{
+    core::{muxing::StreamMuxerBox, transport::ListenerEvent},
+    identity, Transport,
+};
+use log::*;
+
+use std::{
+    any::TypeId,
+    convert::Infallible,
+    io,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+type ListenerUpgrade = Pin<Box<(dyn Future<Output = Result<(PeerId, StreamMuxerBox), io::Error>> + Send + 'static)>>;
+type PeerListener = Pin<Box<dyn Stream<Item = Result<ListenerEvent<ListenerUpgrade, io::Error>, io::Error>> + Send>>;
+
+pub static NUM_LISTENER_EVENT_PROCESSING_ERRORS: AtomicUsize = AtomicUsize::new(0);
+
+lazy_static! {
+    pub static ref LISTEN_ADDRESSES: Arc<RwLock<Vec<Multiaddr>>> = Arc::new(RwLock::new(Vec::new()));
```

Comment:
yea, I see your point, and I am away of this issue regarding `RwLock`s. The listen addresses are only set at initialization, and usually not during the runtime of the node. I think I'll switch to `OnceCell`. Wdyt?

The reason is that multiple task need to have read access to that `Vec` of `Multiaddr`esses. Doing it non-globally would mean passing a copy to each of those tasks managing the synchronization manually. This would increase code complexity both in terms of code lines and fn signatures.